### ISSUE TITLE
Add a "Needs DOI" filter to DOI management page

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#7682]
**Describe the problem you would like to solve**
The new DOI management page supports filtering by several statuses. However, it's not possible to identify all submissions that have not yet been assigned a DOI.

**Describe the solution you'd like**
Add a "Needs DOI" filter to the DOI management page, that will show all submissions in the copyediting/production stages that have at least on enabled DOI object without a DOI.

**Who is asking for this feature?**
Feedback on new DOI management page in #7014.

**Additional information**
cc @ewhanson

**Comment by ewhanson** [comment]
The initial DOI management filters operate only on the 5 DOI statuses (listed below). These should be expanded to more appropriately map to the ways users will interact with the DOI management page.

**Existing filter behaviour**:

Filter | Behaviour
--- | ---
Unregistered | Items with DOIs labeled `STATUS_UNREGISTERED`
Submitted | Items with DOIs labeled `STATUS_SUBMITTED`
Registered | Items with DOIs labeled `STATUS_REGISTERED`
Error | Items with DOIs labeled `STATUS_ERROR`
Stale | Items with DOIs labeled `STATUS_STALE`

**Proposed filter behaviour**:

Filter | Behaviour
--- | ---
Needs DOI | Any published or unpublished item that **1)** has a sub-object without a DOI assigned where **2)** the sub-object's DOI type is enabled (e.g. a galley without a DOI won't appear here if the galley DOI type is not enabled)
Unpublished | An item that **1)** has DOIs assigned for all enabled DOI types, and **2)** has not been published
Unregistered | Items that **1)** has DOIs assigned for  all enabled DOI types, **2)** is published, and **3)** has its DOIs labeled `STATUS_UNREGISTERED`
Submitted | Items with DOIs labeled `STATUS_SUBMITTED`
Registered | Items with DOIs labeled `STATUS_REGISTERED`
Error | Items with DOIs labeled `STATUS_ERROR`
Stale | Items with DOIs labeled `STATUS_STALE`

@NateWr, does this match the use cases we've discussed? It would also be helpful to articulate these behaviours somewhere, too. Would this best live in a help section within OJS/OMP/OPS, or in the documentation?

**Comment by ewhanson** [comment]
Hey @NateWr, I've created draft PRs for this change in OJS. Could you have a look? If the above logic sounds good, I'll port it to OMP and OPS. Thanks!

PRs:
- `ojs`: pkp/ojs#3308
- `ui-library`: pkp/ui-library#180
- `pkp-lib`: #7717

**Comment by NateWr** [comment]
@ewhanson Yes, those use-cases sound right. I agree we should document this. In the near-term, I 'd suggest the [DOI Plugin Guide](https://docs.pkp.sfu.ca/doi-plugin/en/) or the [Crossref Plugin Guide](https://docs.pkp.sfu.ca/crossref-ojs-manual/en/). Maybe bring it to the #documentation group in Slack.

**Comment by ewhanson** [comment]
Hi @NateWr, I've ported the changes to OMP and OPS, and the changes are ready for review. Could you have a look? Thanks.

PRs:
- `pkp-lib`: #7717
- `ui-library`: pkp/ui-library#180
- `ojs`: pkp/ojs#3308
- `omp`: pkp/omp#1070
- `ops`: pkp/ops#238

**Comment by NateWr** [comment]
Looks good! I left a few comments. In addition, could you write a test for these filters? Basically, test that when clicking each filter you see the submissions you expect to. (You probably won't be able to test Error or Stale very easily. Those can wait.)

**Comment by ewhanson** [comment]
Hi @asmecher, I've rebased these and they should be good to go now. Thanks!

PRs:

- `pkp-lib`: #7717
- `ui-library`: pkp/ui-library#180
- `ojs`: pkp/ojs#3308
- `omp`: pkp/omp#1070
- `ops`: pkp/ops#238

**Comment by ewhanson** [comment]
Closing this should also close #7684.

**Comment by ewhanson** [comment]
Last issue with OJS tests now sorted, and all tests are passing.

**Comment by asmecher** [comment]
Just took a quick look -- I think there's one unlocalized string in the pkp-lib PR, otherwise everything looks ready to me.

**Comment by ewhanson** [comment]
@asmecher, I've come across a seemingly unrelated bug in OMP with PHP 8.1 and MySQL with the submission file genre selection process. Would you like me to create a separate issue for it, or try and tackle it before getting this merged? Thanks!

**Comment by NateWr** [comment]
I found two bugs but neither of them was related to your code! :laughing: All merged to `main`, thanks!

**Comment by ewhanson** [comment]
Hey @emmauhl, tagging you for documentation-related updates. 

The DOI refactor, among other things, add new ways to filter DOIs that have specific behaviour that would be good to document. The behaviour is as follows:

| Filter | Behaviour |
| --- | --- |
| Needs DOI | Any published or unpublished item that **1)** has a sub-object without a DOI assigned where **2)** the sub-object's DOI type is enabled (e.g. a galley without a DOI won't appear here if the galley DOI type is not enabled)
| Unpublished| An item that **1)** has DOIs assigned for all enabled DOI types, and **2)** has not been published
| Unregistered | Items that **1)** has DOIs assigned for all enabled DOI types, **2)** is published, and **3)** has its DOIs labeled `STATUS_UNREGISTERED`
| Submitted | Items with DOIs labeled `STATUS_SUBMITTED`
| Registered | Items with DOIs labeled `STATUS_REGISTERED`
| Error | Items with DOIs labeled `STATUS_ERROR`
| Stale | Items with DOIs labeled `STATUS_STALE`

**Comment by asmecher** [comment]
> @asmecher, I've come across a seemingly unrelated bug in OMP with PHP 8.1 and MySQL with the submission file genre selection process. Would you like me to create a separate issue for it, or try and tackle it before getting this merged? Thanks!

@ewhanson, sorry, I've just noticed this message. Is it still needing to be filed and resolved or did it get covered here?

**Comment by ewhanson** [comment]
> @ewhanson, sorry, I've just noticed this message. Is it still needing to be filed and resolved or did it get covered here?

@asmecher, nope, it's been resolved. This was the file genre "other" issue with the jQuery validation cutting in that we dug into a bit and Nate fixed.
